### PR TITLE
Shorten zq and leibpilem1; add nn0onn

### PR DESCRIPTION
**Pull request by icecream17**
Two theorems I noticed could be shortened, before starting onto the generalize zsqrtelqelz and subtheorems

As a sidenote, dffltz can be improved by eldifsnneq (very recent) and ss2ralv (still in #3097). But I wonder if a temporary definition like ``F = ( n e. ( ZZ>= ` 3 ) , s e. ~P CC |-> { <. <. x , y >. , z >. | ( ( x ^ n ) + ( y ^ n ) ) = ( z ^ n ) } )`` would be worth it.

### CODE REVIEW

**Review by wlammen — APPROVED** [review]

File: set.mm [hunk 1]
```diff
@@ -359471,10 +359479,20 @@ already know is total (except at ` 0 ` ).  There are branch points at
       $.
   $}
 
+  $( Lemma for ~ leibpi .  (Contributed by Mario Carneiro, 7-Apr-2015.)  (Proof
+     shortened by Steven Nguyen, 23-Mar-2023.) $)
+  leibpilem1 $p |- ( ( N e. NN0 /\ ( -. N = 0 /\ -. 2 || N ) ) ->
+    ( N e. NN /\ ( ( N - 1 ) / 2 ) e. NN0 ) ) $=
```
Comment (benjub) [inline set.mm:359485]: The proof as well as the usages seem to show that there is not much gain in having these two theorems bundled like that into one. Also, `-. N = 0` is implied. I think two theorems (and not merely lemmas) would be justified:
```
|- ( ( N e. NN0 /\ -. 2 || N ) -> N e. NN ) $=
|- ( ( N e. NN0 /\ -. 2 || N ) -> ( N - 1 ) / 2 ) e. NN0 ) $=
```
and maybe also
```
|- ( ( N e. NN0 /\ -. 2 || N ) -> ( N + 1 ) / 2 ) e. NN ) $=
```

**Comment by wlammen** [comment]
Is there a reason to keep this PR on hold any longer?  The two modified theorems are replacements for the better, both with shortened proofs, and none based on more axioms than the original.  While there might be headroom for further improvements or considerations in the big picture, I don't see why this cannot be postponed until this is merged first.

**Comment by icecream17** [comment]
(I was actually implementing benjub's first two theorems, although the point still stands)

Edit: Actually, given that the second theorem is just nn0oddm1d2 + biimpa, I'm only implementing nn0onn (this commit was made yesterday so it appears above wlammen's comment but it was actually published a few minutes after this comment)

**Comment by benjub** [comment]
Thanks. Now that we have both ~nn0onn and ~nn0oddm1d2, and looking at the two usages of ~leibpilem1, I think that ~leibpilem1 can be removed (i.e., in its two usages, replace it with ~nn0onn or ~nn0oddm1d2 with possibly some glue).

File: set.mm [hunk 2]
```diff
@@ -175578,6 +175586,48 @@ equal to the half of the predecessor of the odd number (which is an even
       EFGZWFWKUDVTWBYGWAWCABUIVQCWEUCVRVS $.
   $}
 
+  $( 0 is even.  (Contributed by AV, 11-Feb-2020.)  (Revised by AV,
```
Comment (benjub) [inline set.mm:175589]: for consistency with the following comments:
>  $( 2 divides 0.  That means 0 is even.

File: set.mm [hunk 3]
```diff
@@ -175578,6 +175586,48 @@ equal to the half of the predecessor of the odd number (which is an even
...
+    wi 2z mp2an mto ) ABCDZABEDZBAFDUCGHBAIJKLAMNBONUBUCRSPABQTUA $.
+
+  $( 2 does not divide -1.  That means -1 is odd.  (Contributed by AV,
+     15-Aug-2021.) $)
+  n2dvdsm1 $p |- -. 2 || -u 1 $=
+    ( c2 c1 cneg cdvds wbr wn caddc co z0even ax-1cn 1pneg1e0 addcomli breqtrri
+    cc0 neg1cn cz wcel wb neg1z oddp1even ax-mp mpbir ) ABCZDEFZAUCBGHZDEZANUED
+    IBUCNJOKLMUCPQUDUFRSUCTUAUB $.
+
+  $( 2 is even.  (Contributed by AV, 12-Feb-2020.)  (Revised by AV,
```
Comment (benjub) [inline set.mm:175607]: > $( 2 divides 2. That means 2 is even.

File: set.mm [hunk 4]
```diff
@@ -175578,6 +175586,48 @@ equal to the half of the predecessor of the odd number (which is an even
...
+     23-Jun-2021.) $)
+  z2even $p |- 2 || 2 $=
+    ( c2 cz wcel cdvds wbr 2z iddvds ax-mp ) ABCAADEFAGH $.
+
+  $( 2 does not divide 3, i.e. 3 is an odd number.  (Contributed by AV,
```
Comment (benjub) [inline set.mm:175612]:
```suggestion
  $( 2 does not divide 3. That means 3 is odd.  (Contributed by AV,
```

File: set.mm [hunk 5]
```diff
@@ -175578,6 +175586,48 @@ equal to the half of the predecessor of the odd number (which is an even
...
+     28-Feb-2021.) $)
+  n2dvds3 $p |- -. 2 || 3 $=
+    ( c2 c3 cdvds wbr wn c1 cmin co cz 2z iddvds ax-mp 3m1e2 breqtrri oddm1even
+    wcel wb 3z mpbir ) ABCDEZABFGHZCDZAAUACAIPAACDJAKLMNBIPTUBQRBOLS $.
+
+  $( 4 is an even number.  (Contributed by AV, 23-Jul-2020.)  (Revised by AV,
```
Comment (benjub) [inline set.mm:175618]: $( 2 divides 4. That means 4 is even.

File: set.mm [hunk 6]
```diff
@@ -175578,6 +175586,48 @@ equal to the half of the predecessor of the odd number (which is an even
...
+     4-Jul-2021.) $)
+  z4even $p |- 2 || 4 $=
+    ( c2 cmul co c4 cdvds cz wcel wbr 2z dvdsmul1 mp2an 2t2e4 breqtri ) AAABCZD
+    EAFGZOANEHIIAAJKLM $.
+
+  $( An integer which is divisible by 4 is an even integer.  (Contributed by
```
Comment (benjub) [inline set.mm:175624]: > $( An integer which is divisible by 4 is divisible by 2, that is, is even.

**Comment by icecream17** [comment]
Would there be leibpilem2OLD and leibpiOLD?

This could be more justified if I rewrite leibpi* to not include the redundant `k = 0`.
Additionally, both times leibpilem1 is used, it's really used to prove 
`(𝑘 ∈ NN0 → if((𝑘 = 0 ∨ 2 ∥ 𝑘), 0, ((-1↑((𝑘 − 1) / 2)) / 𝑘)) ∈ ℂ)`, which suggests a completely separate lemma, although some of the intermediate steps are used for other closures... But this pr is already getting much bigger than expected.

**Review by avekens — APPROVED** [review]
For me, this PR is fine. The revision of the comments as proposed by @benjub would be OK for me, too.

**Comment by benjub** [comment]
> For me, this PR is fine. The revision of the comments as proposed by @benjub would be OK for me, too.

Fine for me. I made some suggestions but nothing is blocking.

File: set.mm (hunk 2 above)
Comment (icecream17) [reply set.mm:175589]: Alright these suggestions should be applied now

**Comment by jkingdon** [comment]
There seems to be a consensus this is ready to merge. Feel free to make follow up pull requests if there are other things people want to change.

**Pull request merged**